terraform/iam.tf

```diff
@@ -0,0 +1,146 @@
+# iam roles
+resource "aws_iam_role" "app-ec2-role" {
+  name               = "app-ec2-role"
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+
+}
+
+resource "aws_iam_instance_profile" "app-ec2-role" {
+  name = "app-ec2-role"
+  role = aws_iam_role.app-ec2-role.name
+}
+
+# service
+resource "aws_iam_role" "elasticbeanstalk-service-role" {
+  name = "elasticbeanstalk-service-role"
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "elasticbeanstalk.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+
+}
+
+resource "aws_iam_role" "iam_for_lambda" {
+  name = "iam_for_lambda"
```

Comment:
Can you prefix all the name. Something like **upgrade_** so that all the resource can be identified from console and no conflict with existing resource.

Reply:
It would be better if we have a variable for prefixing